### PR TITLE
Add timeout and max_memory limit support

### LINKED ISSUE #63: Limit resource (execution time and memory usages)
Awesome lib and easy to install thanks to the wheels build :smile: . I have a question about limiting resources for the executes Javascript. I want to run user inputted script but want to limit the execution time and memory usages.

What is the best approach to do this, Thanks.

### PR DESCRIPTION
**Pull request by messense**
Closes #63 

Can't build v8 on macOS 10.14, so let's see what Travis CI says...

### CODE REVIEW

File: py_mini_racer/extension/mini_racer_extension.cc [hunk 1]
```diff
@@ -497,7 +540,7 @@ ContextInfo *MiniRacer_init_context()
 
 static BinaryValue* MiniRacer_eval_context_unsafe(
         ContextInfo *context_info,
-        char *utf_str, int   str_len)
+        char *utf_str, int str_len, long timeout, size_t max_memory)
```
Comment (Taiki-San) [inline py_mini_racer/extension/mini_racer_extension.cc]: Maybe break the line after `str_len` to keep the line length consistent?

**Comment by messense** [comment]
Addressed review comments.

**Comment by Taiki-San** [comment]
Looks good to me. Good catch for checking for None when freeing `res`

**Comment by Taiki-San** [comment]
@messense I just tried to run the tests on my computer and the two new tests appear not to pass. No idea why they didn't made Travis fail.  
Specifically, the memory test appear to fail when `concat`-ing the array. The runtime test appear to be running infinitely.

If you want to build PMR on your Mac, you can either run the build twice (the second time won't rebuild V8 so will be very short), or set the env variable mentioned in the readme.

If the patch isn't trivial, I'll revert the PR and let you submit a new one.

**Comment by messense** [comment]
@Taiki-San Can you try a clean git clone, build then test?

**Comment by Taiki-San** [comment]
@messense A fresh build and some tweaks to my environment fixed the issue. Sorry for the inconvenience.

**Pull request merged**